src/Pass/IframeBrightCoveTagTransformPass.php

```diff
@@ -0,0 +1,160 @@
+            $dom_el = $el->get(0);
+            if (!$this->isBrighCoveIframe($el)) {
+                continue;
+            }
+
+            $lineno = $this->getLineNo($dom_el);
+            $context_string = $this->getContextString($dom_el);
+            $brightcove_code = $this->getBrightcoveCode($el);
+
+            // If we couldnt find a BrighCove videoid then we abort
+            if (empty($brightcove_code)) {
+                continue;
+            }
+
+            if ($el->hasAttr('class')) {
+                $class_attr = $el->attr('class');
+            }
+
+            /** @var \DOMElement $new_dom_el */
+            $el->after("<amp-brightcove data-videoid=\"$brightcove_code\" layout=\"responsive\"></amp-brightcove>");
+            $new_el = $el->next();
+            $new_dom_el = $new_el->get(0);
+            if (!empty($class_attr)) {
+                $new_el->attr('class', $class_attr);
+            }
+            $this->setStandardAttributesFrom($el, $new_el, self::DEFAULT_VIDEO_WIDTH, self::DEFAULT_VIDEO_HEIGHT, self::DEFAULT_ASPECT_RATIO);
+            $this->setBrightcoveAttributesFrom($el, $new_el);
+
+            // Remove the iframe and its children
+            $el->removeChildren()->remove();
+            $this->addActionTaken(new ActionTakenLine('iframe', ActionTakenType::BRIGHTCOVE_IFRAME_CONVERTED, $lineno, $context_string));
+            $this->context->addLineAssociation($new_dom_el, $lineno);
+        }
+
+        return $this->transformations;
+    }
+
+    /**
+     * @param DOMQuery $el
+     * @return bool
+     */
+    protected function isBrighCoveIframe(DOMQuery $el)
+    {
+        $href = $el->attr('src');
+        if (empty($href)) {
+            return false;
+        }
+
+        if (preg_match('&(*UTF8)(brighcove\.com)&i', $href)) {
+            return true;
+        }
+
+        return false;
+    }
+
+    /**
+     *
+     * Get the brightcove videoid
+     *
+     * @param DOMQuery $el
+     * @return string
+     */
+    protected function getBrightcoveCode(DOMQuery $el)
+    {
+        $matches = [];
+        $brightcove_code = '';
+        $href = $el->attr('src');
+
+        // @todo there seem to be a lot of ways to embed a brightcove video. We probably need to capture all patterns here
+        // The next one is the embed code that brightcove gives you
+        if (preg_match('&(*UTF8)/embed/([^/?]+)&i', $href, $matches)) {
+            if (!empty($matches[1])) {
+                $brightcove_code = $matches[1];
+            }
+        }
+        elseif (preg_match('&(*UTF8)brightcove/([^/?]+)&i', $href, $matches)) {
+            if (!empty($matches[1])) {
+                $brightcove_code = $matches[1];
+            }
+        }
+        elseif (preg_match('!(*UTF8)watch\?v=([^&]+)!i', $href, $matches)) {
+            if (!empty($matches[1])) {
+                $brightcove_code = $matches[1];
+            }
+        }
+
+        return $brightcove_code;
```

Comment:
You need to make sure to return valid HTML code here..
wrap the return value in `htmlspecialchars`

See example here:
https://github.com/Lullabot/amp-library/blob/master/src/Pass/IframeYouTubeTagTransformPass.php#L134